src/main/jhipster/src/main/webapp/app/file-manager/file-manager.component.ts

```diff
@@ -161,7 +161,26 @@ export class FileManagerComponent {
         }
     }
 
+    validateFileNameDuplicated() {
+        let filesObservations = [];
+        if (this.variable && this.variable.id) {
+            const variableId = Number(this.variable.id);
+            filesObservations = this.fileMetadataList.filter((fileMetadata) => //
+                fileMetadata.variables.length !== 0 && Number(fileMetadata.variables[0].id) === variableId);
+        } else {
+            filesObservations = this.fileMetadataList.filter((fileMetadata) => //
+                fileMetadata.name === this.file.name && fileMetadata.variables.length === 0);
+        }
+
+        return filesObservations.filter((fileMetadata) => fileMetadata.name === this.file.name).length !== 0;
+    }
+
```

Comment:
I think this can be simplified, and also it would be better to prevent the overwrite also regardless of the variable, because the file key to overwritte the storage entry depends only on the obs unit uuid.
In other words:
```typescript
   validateFileNameDuplicated() {
        let filesObservations = [];
        filesObservations = this.fileMetadataList.filter((fileMetadata) => fileMetadata.name === this.file.name);
        return filesObservations.length !== 0;
    }
```

Reply:
@nahuel-soldevilla I modified the logic to search if exists the filename using the listFileMetadata service.